src/animation_loader.rs

```diff
@@ -1,30 +1,46 @@
-use bevy::prelude::*;
+use bevy::{
+    asset::{Asset, Handle},
+    ecs::system::SystemParamItem,
+    prelude::*,
+    render::render_asset::{self, RenderAsset, RenderAssetPlugin, RenderAssetUsages},
+};
 use bvh_anim::{errors::LoadError, Bvh};
 use std::{fs, io::BufReader};
 
 pub struct AnimationLoaderPlugin;
 
 impl Plugin for AnimationLoaderPlugin {
     fn build(&self, app: &mut App) {
-        app.add_systems(Startup, store_bvh);
-        app.insert_resource(BvhData::new());
+        app.add_plugins(RenderAssetPlugin::<BvhAsset>::default())
+            .init_asset::<BvhAsset>()
+            .add_systems(Startup, store_bvh);
     }
 }
 
-#[derive(Resource)]
-pub struct BvhData {
-    pub bvh_animation: Vec<Bvh>,
+#[derive(Component, Clone, Default)]
+pub struct BvhHandle {
+    pub bvh_assets: Handle<BvhAsset>,
 }
 
-impl BvhData {
-    pub fn new() -> Self {
-        Self {
-            bvh_animation: Vec::new(),
-        }
+#[derive(Asset, TypePath, Clone, Default)]
+pub struct BvhAsset {
+    pub asset: Bvh,
+}
+
+impl RenderAsset for BvhAsset {
+    type PreparedAsset = Self;
+
+    type Param = ();
+
+    fn asset_usage(&self) -> RenderAssetUsages {
+        RenderAssetUsages::MAIN_WORLD | RenderAssetUsages::RENDER_WORLD
     }
 
-    pub fn get_bvh_animation_data(&self, animation_data_index: usize) -> &Bvh {
-        return &self.bvh_animation[animation_data_index];
+    fn prepare_asset(
+        self,
+        _param: &mut SystemParamItem<Self::Param>,
+    ) -> Result<Self::PreparedAsset, render_asset::PrepareAssetError<Self>> {
+        Ok(self)
```

Comment:
I don't think we need it to be a `RenderAsset` as we are not rendering the Bvh data..